src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h

```diff
@@ -1557,45 +1565,63 @@ PUBLIC_API STATUS signalingClientGetIceConfigInfoCount(SIGNALING_CLIENT_HANDLE,
  * IMPORTANT: The returned pointer to the ICE configuration information object points to internal structures
  * and its contents should not be modified.
  *
- * @param SIGNALING_CLIENT_HANDLE
+ * @param[in] SIGNALING_CLIENT_HANDLE Signaling client handle
  * @param[in] UINT32 Index of the ICE configuration information object to retrieve
  * @param[out] PIceConfigInfo The pointer to the ICE configuration information object
  *
  * @return STATUS code of execution. STATUS_SUCCESS on success
  */
 PUBLIC_API STATUS signalingClientGetIceConfigInfo(SIGNALING_CLIENT_HANDLE, UINT32, PIceConfigInfo*);
 
-
 /**
  * @brief Connects the signaling client to the socket in order to send/receive messages.
  *
  * NOTE: The call will succeed only when the signaling client is in a ready state.
- * @param SIGNALING_CLIENT_HANDLE
  *
- * @return STATUS function execution status. STATUS_SUCCESS on success
+ * @param[in] SIGNALING_CLIENT_HANDLE Signaling client handle
+ *
+ * @return STATUS code of the execution. STATUS_SUCCESS on success
  */
 PUBLIC_API STATUS signalingClientConnectSync(SIGNALING_CLIENT_HANDLE);
 
 /*
- * Gets the Signaling client current state.
+ * @brief Gets the Signaling client current state.
  *
- * @param - SIGNALING_CLIENT_HANDLE - IN - Signaling client handle
- * @param - PSIGNALING_CLIENT_STATE - OUT - Current state of the signaling client as an UINT32 enum
+ * @param[in] SIGNALING_CLIENT_HANDLE Signaling client handle
+ * @param[out] PSIGNALING_CLIENT_STATE Current state of the signaling client as an UINT32 enum
  *
- * @return - STATUS code of the execution
+ * @return STATUS code of the execution. STATUS_SUCCESS on success
  */
 PUBLIC_API STATUS signalingClientGetCurrentState(SIGNALING_CLIENT_HANDLE, PSIGNALING_CLIENT_STATE);
 
 /*
  * Gets a literal string representing a Signaling client state.
  *
- * @param - SIGNALING_CLIENT_STATE - IN - Signaling client state
- * @param - PCHAR* - OUT - Read only string representing the state
+ * @param[in] SIGNALING_CLIENT_HANDLE Signaling client handle
+ * @param[out] PCHAR* Read only string representing the state
  *
- * @return - STATUS code of the execution
+ * @return STATUS code of the execution. STATUS_SUCCESS on success
  */
 PUBLIC_API STATUS signalingClientGetStateString(SIGNALING_CLIENT_STATE, PCHAR*);
 
+/**
+ * @brief Deletes the signaling channel referenced by SIGNALING_CLIENT_HANDLE
+ *
+ * NOTE: The function is intended to be used to clean up the backend resources and
+ * as such should be called at the end of the lifecycle of the signaling channel resource.
+ * Attempting to connect to the channel or send a message will result in an
+ * error or an unpredictable results after this call.
+ *
+ * NOTE: The call transitions the signaling client state machine to a terminal state
```

Comment:
so what happens when signalingClientDeleteSync fails but the signaling channel is not successfully deleted? Is it still possible to delete it using the sdk?

Reply:
when delete channel is called, can we just terminate the state machine and wait on delete api to be finished? The delete channel operation seems orthogonal to any of the state machine operations. This way in the state machine we dont need to worry if about delete happening.

Reply:
The state is a terminal state. There isn't much left for the application to do but to either issue the delete call again or free the client.

We can't simply terminate the state machine as the delete needs the state machine for a couple of reasons - 1) in case of the stale credentials we will need to iterate to get token state and 2) in case of the delete failing with 400 we might have a bad version in which case we need to call describe.